### PR TITLE
feat: add prepare target, option to depend on prepare and start symlinking to bin dir

### PR DESCRIPTION
**Pull request by viktorvoltaire**
- chore: add prepare function to each target
- feat: add option to depend on prepare

### CODE REVIEW

File: mgtool/build.go [hunk 1]
```diff
@@ -12,21 +12,36 @@ import (
...
 	}
 	goPkgVer := fmt.Sprintf("%s@%s", goPkg, version)
 	os.Setenv("GOBIN", filepath.Dir(executable))
 	logr.FromContextOrDiscard(ctx).Info(fmt.Sprintf("Building %s...", goPkgVer))
 	if err := sh.RunV("go", "install", goPkgVer); err != nil {
 		return "", err
 	}
-	return executable, nil
+	if err := createSymlink(executable, symlink); err != nil {
+		return "", err
```
Comment (Edholm) [inline mgtool/build.go:32]: maybe wrap this err to make it easier to troubleshoot:

```suggestion
		return "", fmt.Errorf("failed to create symlink: %w", err)
```

**Review by Edholm — APPROVED** [review]
Looks good! :+1: 

What is the reason for the change though? :thinking:

**Comment by odsod** [comment]
@Edholm We want to be able to prepare tools without using them from mage - for example when they are used by Buf, or `go generate`.

File: mgtool/build.go (hunk 1 above)
Comment (odsod) [reply mgtool/build.go:32]: IMO the `createSymlink` function should talk about failure to add symlink

File: targets/mgconvco/targets.go [hunk 2]
```diff
@@ -21,6 +21,12 @@ const version = "0.3.7"
 // nolint: gochecknoglobals
 var executable string
 
+type Prepare mgtool.Prepare
+
+func (Prepare) ConvcoCheck() {
```
Comment (odsod) [inline targets/mgconvco/targets.go:26]: should be `Convco`

File: mgtool/configuration.go [hunk 3]
```diff
@@ -3,13 +3,17 @@ package mgtool
...
+
+	"github.com/magefile/mage/mg"
 )
 
 const (
 	AMD64 = "amd64"
 	X8664 = "x86_64"
 )
 
+type Prepare mg.Namespace
```
Comment (odsod) [inline mgtool/configuration.go:15]: is this used to ignore outputting to Makefiles? maybe add as comment?

File: targets/mgsops/targets.go [hunk 4]
```diff
@@ -19,15 +19,21 @@ const version = "3.7.1"
 // nolint: gochecknoglobals
 var executable string
 
+type Prepare mgtool.Prepare
+
+func (Prepare) Sops() {
+	mg.Deps(prepare)
```
Comment (odsod) [inline targets/mgsops/targets.go:25]: could we just inline the `prepare` function here?

File: mgtool/configuration.go (hunk 3 above)
Comment (viktorvoltaire) [reply mgtool/configuration.go:15]: Yeah

File: targets/mgsops/targets.go (hunk 4 above)
Comment (viktorvoltaire) [reply targets/mgsops/targets.go]: Yep, fixed

File: targets/mgconvco/targets.go (hunk 2 above)
Comment (viktorvoltaire) [reply targets/mgconvco/targets.go]: Fixed

File: mgtool/build.go (hunk 1 above)
Comment (viktorvoltaire) [reply mgtool/build.go:32]: The error I got, from when trying to create a symlink when one already existed was clear enough. I think we can stick with what we got

File: targets/mgsops/targets.go (hunk 4 above)
Comment (odsod) [reply targets/mgsops/targets.go]: Might as well inline all prepare functions?

**Review by odsod — APPROVED** [review]
:rocket:

**Pull request merged**